consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -1028,8 +1040,8 @@ impl PruningProofManager {
                     block_at_depth_2m
                 };
 
-                // new root is expected to be always an ancestor of old root because new root takes a safety margin
-                assert!(self.reachability_service.is_dag_ancestor_of(root, old_root));
+                // new root is expected to be always an ancestor of depth_based_root because new root takes a safety margin
+                assert!(self.reachability_service.is_dag_ancestor_of(root, depth_based_root));
```

Comment:
I don't think this assertion is guaranteed to be true. What if we get to [this](https://github.com/kaspanet/rusty-kaspa/pull/494/files/3e249f91cd3f5fadc1a1ee647a352cc5fa7e3d31..46dbac30a78fa48b503621d8970e019e9a45aa31#diff-593674986e16100c0c323e07c342dc26ddf962b1efce504ce245975d817f5922R952) branch?

Reply:
If it gets to that branch, then likely that calculated root == depth_based_root. The GD store `block_at_depth` uses is one that was constructed through `find_sufficient_root`.

The GD store is constructed by BFS forward traversing from calculated root to selected tip in `find_sufficient_root`, ignoring blocks that are not in the past of selected tip. So all blocks in this GD store must be in the future of root. More importantly, all selected parent chain from any block (other than root) in this store must lead back to root.

`depth_based_root` is calculated by traversing the selected parent chain from selected tip back to root. The furthest back that this backward chain traversal can go is root.

So at all times, root must be an ancestor of or equal to depth_based_root.

Reply:
Ok, so IIUC this branch can only cause block_at_depth to fail, right?